modules/videoio/src/videoio_registry.cpp

```diff
@@ -333,7 +333,24 @@ class VideoBackendRegistry
 
 } // namespace
 
-namespace videoio_registry {
+namespace videoio_registry { namespace internal {
+    const VideoBackendInfo& getBackendInfo(VideoCaptureAPIs api)
+    {
+        if (api == CAP_ANY)
+            CV_Error(Error::StsError, "Wrong backend ID: CAP_ANY");
```

Comment:
`CV_Assert(api != CAP_ANY);` is preferable (this check is an internal assumption)

Reply:
ic